### PR TITLE
Search results changes no tests

### PR DESCRIPTION
**Pull request by julian1**

### CODE REVIEW

**Comment by anguss00** [comment]
I'll review

File: web-app/js/portal/data/TermClassification.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        extractCategories(record, 0, result);
+        return result;
+    }
+
+    var constructor = Ext.data.Record.create([
+        { name: 'categories', convert: extractDimension }
+    ]);
+
+    return constructor;
+}();
```
Comment (anguss00) [inline web-app/js/portal/data/TermClassification.js]: extra bracket here

File: web-app/js/portal/search/FacetedSearchResultsDataView.js [hunk 2]
```diff
@@ -63,6 +61,22 @@ Portal.search.FacetedSearchResultsDataView = Ext.extend(Ext.DataView, {
...
+                },
+                getTitle: function(values) {
+                    var broader = [];  
+                    Ext.each(values.parameter, function(param) {
+                        var broaderTerms = this.classificationStore.getBroaderTerms(param, 2, 'Measured parameter');
+                        if(broaderTerms.length > 0) { 
+                            broader = broader.concat(broaderTerms); 
+                        }
+                        else {
+                            console.log("Could not find broader term for parameter '" + param  + "'");
```
Comment (anguss00) [inline web-app/js/portal/search/FacetedSearchResultsDataView.js]: I think we should remove the console logs as they fire off a lot of responses on load... Maybe they should be error logged in the console instead if logging is required

File: web-app/js/portal/search/FacetedSearchResultsDataView.js [hunk 3]
```diff
@@ -141,13 +154,23 @@ Portal.search.FacetedSearchResultsDataView = Ext.extend(Ext.DataView, {
...
+                return !pos || item != broader[pos - 1];
+            });
             return template.apply({
                 "label": label,
-                "value": platform
+                "value": broader.join( ', ')
             });
         }
+        else {
+            console.log("Could not find broader term for platform '" + platform  + "'");
```
Comment (anguss00) [inline web-app/js/portal/search/FacetedSearchResultsDataView.js]: Logging here as above

File: web-app/js/portal/search/FacetedSearchResultsDataView.js (hunk 2 above)
Comment (anguss00) [inline web-app/js/portal/search/FacetedSearchResultsDataView.js]: so the code I was thinking of is log.error("Could not find broader term for parameter '" + param + "'")

File: web-app/js/portal/data/TermClassification.js (hunk 1 above)
Comment (julian1) [inline web-app/js/portal/data/TermClassification.js]: @anguss00 do you mean the (); at the end? It's needed (at least it doesn't work without it) and follows the same pattern as the corresponding GeonetworkRecord store code.

Comment (anguss00) [inline web-app/js/portal/data/TermClassification.js]: Ah right! My mistake, it's just a different format to how we normally close classes. All good then

File: web-app/js/portal/search/FacetedSearchResultsDataView.js (hunk 2 above)
Comment (julian1) [inline web-app/js/portal/search/FacetedSearchResultsDataView.js]: I think I'll comment them. Just to keep a record - these are the missing parameter/platform terms.

window.controllers is deprecated. Do not use it for UA detection. OpenLayers.js:1171
"Could not find broader term for parameter 'Speed (over ground) of measurement platform'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Direction of motion (over ground) of measurement platform'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Orientation (pitch) of measurement platform'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Orientation (roll angle) of measurement platform'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Direction of motion (over ground) of measurement platform'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Height above sensor'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Depth below surface of the water body'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Correlation in magnitude of the particle distribution in the water body'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Percentage of good three beam solutions (one beam rejected) in ADCP bin'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Percentage of good transformations (error velocity threshold not exceeded) in ADCP bin'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Percentage of measurements where more than one beam was bad in ADCP bin'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Percentage of measurements with four beam solutions in ADCP bin'" FacetedSearchResultsDataView.js:73
"Could not find broader term for parameter 'Depth below surface of the water body'" FacetedSearchResultsDataView.js:73
"Could not find broader term for platform ''" FacetedSearchResultsDataView.js:172
"Could not find broader term for parameter 'Depth below surface of the water body'" FacetedSearchResultsDataView.js:73

File: web-app/js/portal/search/FacetedSearchResultsDataView.js [hunk 4]
```diff
@@ -63,6 +61,19 @@ Portal.search.FacetedSearchResultsDataView = Ext.extend(Ext.DataView, {
...
+                },
+                getTitle: function(values) {
+                    var broader = [];  
+                    Ext.each(values.parameter, function(param) {
+                        var broaderTerms = this.classificationStore.getBroaderTerms(param, 2, 'Measured parameter');
+                        if(broaderTerms.length > 0) { 
+                            broader = broader.concat(broaderTerms); 
+                        }
+                    }, this);
+                    broader = broader.sort();
```
Comment (julian1) [inline web-app/js/portal/search/FacetedSearchResultsDataView.js]: @anguss00 removed logging as discussed.

**Comment by anguss00** [comment]
:+1:

**Pull request merged**